api/src/paths/project/{projectId}/attachments/list.test.ts

```diff
@@ -0,0 +1,139 @@
+import chai, { expect } from 'chai';
+import { describe } from 'mocha';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import * as listAttachments from './list';
+import * as db from '../../../../database/db';
+import * as project_attachments_queries from '../../../../queries/project/project-attachments-queries';
+import SQL from 'sql-template-strings';
+
+chai.use(sinonChai);
+
+describe('lists the project attachments', () => {
+  const dbConnectionObj = {
+    systemUserId: () => {
+      return null;
+    },
+    open: async () => {
+      // do nothing
+    },
+    release: () => {
+      // do nothing
+    },
+    commit: async () => {
+      // do nothing
+    },
+    rollback: async () => {
+      // do nothing
+    },
+    query: async () => {
+      // do nothing
+    }
+  };
+
+  const sampleReq = {
+    keycloak_token: {},
+    body: {},
+    params: {
+      projectId: 1
+    }
+  } as any;
+
+  let actualResult: any = null;
+
+  const sampleRes = {
+    status: () => {
+      return {
+        json: (result: any) => {
+          actualResult = result;
+        }
+      };
+    }
+  };
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should throw a 400 error when no sql statement returned for getProjectAttachmentsSQL', async () => {
+    sinon.stub(db, 'getDBConnection').returns({
+      ...dbConnectionObj,
+      systemUserId: () => {
+        return 20;
+      }
+    });
+
+    sinon.stub(project_attachments_queries, 'getProjectAttachmentsSQL').returns(null);
+
+    try {
+      const result = listAttachments.getAttachments();
+
+      await result(sampleReq, (null as unknown) as any, (null as unknown) as any);
+      expect.fail();
+    } catch (actualError) {
+      expect(actualError.status).to.equal(400);
+      expect(actualError.message).to.equal('Failed to build SQL get statement');
+    }
+  });
+
+  it('should return a list of project attachments on success', async () => {
+    const mockQuery = sinon.stub();
+
+    mockQuery.resolves({ rows: [{ id: 13, fileName: 'name1', lastModified: 'now', size: 50 }] });
+
+    sinon.stub(db, 'getDBConnection').returns({
+      ...dbConnectionObj,
+      systemUserId: () => {
+        return 20;
+      },
+      query: mockQuery
+    });
+
+    sinon.stub(project_attachments_queries, 'getProjectAttachmentsSQL').returns(SQL`something`);
+
+    const result = listAttachments.getAttachments();
+
+    await result(sampleReq, sampleRes as any, (null as unknown) as any);
+
+    expect(actualResult).to.not.be.null;
```

Comment:
should be able to assert the actual value of the result, not just that it isnt null

Reply:
Yeah, not super happy with this test.  I seem to not be able to see the id in the actualResult and the others are undefined.

Reply:
should be fixed now